### PR TITLE
TCES-26 Add Client

### PR DESCRIPTION
**Pull request by granthamblin**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Issue #

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

Please delete options that are not applicable.

- [x] My code follows the style guidelines of this project (required)
- [x] I have performed a self-review of my code (required)
- [x] I have linted and formatted my code (required)
- [x] New and existing unit tests pass locally with my changes (required)
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by jordanjanakievski — COMMENTED** [review]
This component is looking great. Functioning as we expect and everything. It will be functioning as a page though so if you could it would be great if you added a respective directory and file in the `pages` folder. Similar to how stuff is set up for `create-user` page.

Call your file `index.jsx` instead of `new-client.jsx`

Here is an example of how you can set up the files so its uniform with everything and following some common practice naming conventions. We often like to use the verbs from CRUD when naming frontend pages and backend related methods since they concisely tell us what is going on. (Create, Read, Update, Delete)

I hope this info is helpful. Just want to help you learn as much as possible
```
| components
| -- create-client-component
------ {Your component files here}
| pages
| -- create-client
------ index.jsx -> your new file for this being a page
```

File: client/src/components/new-client-component/new-client-component.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        <Header>
+          <Typography variant="h3">Add a New Client</Typography>
+          <Typography variant="body1">
+            Input information about the client
+          </Typography>
+        </Header>
+        {clients.map((client) => (
+          <ClientCard key={client.id} />
+        ))}
+        <AddButton onClick={handleAddClient}>+ Add Another Client</AddButton>
```
Comment (DanielDervishi) [inline client/src/components/new-client-component/new-client-component.jsx]: Hovering over this should change the cursor to a pointer.

File: client/src/components/new-client-component/new-client-component.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+} from "@mui/material";
+import { Header, Form, Discard, AddButton } from "./new-client.styles";
+import ClientCard from "./ClientCard";
+
+function NewClient() {
+  const [clients, setClients] = useState([
```
Comment (DanielDervishi) [inline client/src/components/new-client-component/new-client-component.jsx]: I like how you created this state in the parent component. However the data within it isn't being changed along with the fields. instead of mainting fullname, phonenumber and email as a state in each clientcard component, instead use the state you've created here. You could then pass the client data to each of the client cards as well as the setClients method (or some variation of these methods as props). 

The code for rendering a child component might end up looking something like this
`<ClientCard clientData={clients[i]} setClientData = {(newClientData) = setClientData(clients.slice(0, i).concat([newClientData]).concat(clients.slice(i + 1, -1))}`

Basically give each component the ability to change the data associated with it in the parent.

File: tces
Comment (DanielDervishi) [inline tces:1]: remove this module

**Review by DanielDervishi — COMMENTED** [review]
Great work once again Grant! If you have any questions about some of my comments (they may be a bit complicated) please reach out 😄

File: client/src/components/create-client-component/ClientCard.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+TextMaskCustom.propTypes = {
+  name: PropTypes.string.isRequired,
+  onChange: PropTypes.func.isRequired,
+};
+
+function ClientCard({ setClientData, index }) {
+  const [values, setValues] = React.useState({
+    fullName: "",
+    textmask: "(100) 000-0000",
```
Comment (jordanjanakievski) [inline client/src/components/create-client-component/ClientCard.jsx]: You can remove this auto-fill for the phone number

**Review by jordanjanakievski — COMMENTED** [review]
Just the one comment about the phone number masking, otherwise looking good!

Great work!

**Review by DanielDervishi — COMMENTED** [review]
Great work Grant, just be sure to pull from main, set the app.jsx file to match the one on main and resolve any conflicts. After this we should be good to merge

**Review by DanielDervishi — APPROVED** [review]
Great Work Grant!

**Review by jordanjanakievski — APPROVED** [review]
Awesome work! Thanks for those changes Grant!
